server/tso/allocator_manager.go

```diff
@@ -87,6 +92,70 @@ func NewAllocatorManager(m *member.Member, rootPath string, saveInterval time.Du
 	return allocatorManager
 }
 
+// SetLocalTSOConfig receives a `LocalTSOConfig` and write it into etcd to make the whole
+// cluster know the DC-level topology for later Local TSO Allocator campaign.
+func (am *AllocatorManager) SetLocalTSOConfig(localTSOConfig config.LocalTSOConfig) error {
+	memberName := am.member.Member().Name
+	memberID := fmt.Sprint(am.member.ID())
+	log.Info("write dc-location into etcd",
+		zap.String("dc-location", localTSOConfig.DCLocation),
+		zap.String("name", memberName),
+		zap.String("member-id", memberID))
+	if !localTSOConfig.EnableLocalTSO {
+		log.Info("pd server doesn't enable local tso, skip writing dc-location into etcd",
+			zap.String("name", memberName),
+			zap.String("member-id", memberID))
+		return nil
+	}
+	// The key-value pair in etcd will be like: memberName -> dcLocation
+	dcLocationKey := path.Join(am.getLocalTSOConfigPath(), memberName)
```

Comment:
how about writing member id instead? I think name can be changed later.

Reply:
Addressed.